lib/elixir/lib/calendar/datetime.ex

```diff
@@ -1096,8 +1097,8 @@ defmodule DateTime do
 
   """
   @doc since: "1.6.0"
-  @spec truncate(t(), :microsecond | :millisecond | :second) :: t()
-  def truncate(%DateTime{microsecond: microsecond} = datetime, precision) do
+  @spec truncate(Calendar.datetime(), :microsecond | :millisecond | :second) :: t()
```

Comment:
The return type is not valid anymore - it's not guaranteed to always return a `DateTime` struct. Similar situation exists in couple other functions.

Reply:
Good catch. I'll go through them and make sure that `DateTime`s will be returned even if a non-`DateTime` is passed.

Reply:
See https://github.com/elixir-lang/elixir/pull/8448